tests/test_date.py

```diff
@@ -239,59 +406,163 @@ def test_should_return_yesterday(self, date_string):
         param('Avant-hier'),
     ])
     def test_should_return_day_before_yesterday(self, date_string):
-        today = datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0)
-        day_before_yesterday = today - relativedelta(days=2)
-        date_data = self.parser.get_date_data(date_string)
-        self.assertIsNotNone(date_data['date_obj'])
-        self.check_equal(day_before_yesterday.date(),
-                         date_data['date_obj'].date(), date_string)
-
-    def test_should_not_assume_language_too_early(self):
-        date_fixtures = [
-            (u'07/07/2014', datetime(2014, 7, 7)),
-            (u'07.jul.2014 | 12:52', datetime(2014, 7, 7)),
-            (u'07.ago.2014 | 12:52', datetime(2014, 8, 7)),
-            (u'07.feb.2014 | 12:52', datetime(2014, 2, 7)),
-            (u'07.ene.2014 | 12:52', datetime(2014, 1, 7)),
-        ]
-
-        for date_string, correct_date in date_fixtures:
-            date_data = self.parser.get_date_data(date_string)
-            self.assertIsNotNone(date_data['date_obj'],
-                                 "Could not get date for: %s" % date_string)
-            self.check_equal(correct_date.date(),
-                             date_data['date_obj'].date(), date_string)
+        self.given_date_string(date_string)
+        self.given_today(datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0))
+        self.given_day_before_yesterday()
+        self.when_date_data()
+        self.then_should_return_day_before_yesterday_is_not_none()
+        self.then_should_return_day_before_yesterday_is_equal()
+
+    @parameterized.expand([
+        param(u'07/07/2014', datetime(2014, 7, 7)),
+        param(u'07.jul.2014 | 12:52', datetime(2014, 7, 7)),
+        param(u'07.ago.2014 | 12:52', datetime(2014, 8, 7)),
+        param(u'07.feb.2014 | 12:52', datetime(2014, 2, 7)),
+        param(u'07.ene.2014 | 12:52', datetime(2014, 1, 7)),
+    ])
+    def test_should_not_assume_language_too_early(self, date_string, correct_date):
```

Comment:
This test check multiple parse on single date parser, so it needs to be written with same approach as the next test.